dotnet/src/webdriver/Remote/DriverServiceCommandExecutor.cs

```diff
@@ -92,6 +93,16 @@ public HttpCommandExecutor HttpExecutor
         /// <param name="commandToExecute">The command you wish to execute</param>
         /// <returns>A response from the browser</returns>
         public Response Execute(Command commandToExecute)
+        {
+            return Task.Run(() => this.ExecuteAsync(commandToExecute)).GetAwaiter().GetResult();
```

Comment:
This is common piece of code how to resolve `sync-over-async` problem. I propose to introduce internal helper class who can do it "properly".

```csharp
internal class AsyncHelper
    {
        private static readonly TaskFactory _myTaskFactory = new TaskFactory(CancellationToken.None,
            TaskCreationOptions.None, TaskContinuationOptions.None, TaskScheduler.Default);

        public static TResult RunSync<TResult>(Func<Task<TResult>> func)
        {
            return _myTaskFactory.StartNew(() =>
            {
                return func();
            }).Unwrap().ConfigureAwait(false).GetAwaiter().GetResult();
        }

        public static void RunSync(Func<Task> func)
        {
            _myTaskFactory.StartNew(() =>
            {
                return func();
            }).Unwrap().ConfigureAwait(false).GetAwaiter().GetResult();
        }
    }
```

This example came from Microsoft (Visual Studio project).

Reply:
@nvborisenko What's an example of how the helper would get called? Like this?
```
return AsyncHelper. RunSync(() => this.ExecuteAsync(commandToExecute));
```

Reply:
Yes, that's right.